src/journal/managed-journal-file.c

```diff
@@ -508,19 +508,25 @@ int managed_journal_file_open_reliably(
         if ((open_flags & O_ACCMODE) == O_RDONLY)
                 return r;
 
-        if (!(open_flags & O_CREAT))
-                return r;
-
         if (!endswith(fname, ".journal"))
                 return r;
 
         /* The file is corrupted. Rotate it away and try it again (but only once) */
-        log_warning_errno(r, "File %s corrupted or uncleanly shut down, renaming and replacing.", fname);
+        log_warning_errno(r, "File %s corrupted or uncleanly shut down, trying to move it out of the way: %m", fname);
 
         r = journal_file_dispose(AT_FDCWD, fname);
-        if (r < 0)
+        if (r < 0) {
+                log_warning_errno(r, "Failed to move %s out of the way: %m", fname);
+                return r;
+        }
```

Comment:
merge the two lines with `return log_warning_errno(r, …`

Reply:
also, even if the whole PR is merged, you'll log about this warning here twice. here and in the caller where you'll warn again